pkg/ipamd/ipamd.go

```diff
@@ -849,22 +847,16 @@ func (c *IPAMContext) increaseDatastorePool(ctx context.Context) error {
 	if increasedPool {
 		c.updateLastNodeIPPoolAction()
 	} else {
-		// Check if we need to make room for the VPC Resource Controller to attach a trunk ENI
-		reserveSlotForTrunkENI := 0
-		if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
-			reserveSlotForTrunkENI = 1
-		}
 		// If we did not add any IPs, try to allocate an ENI.
-		if c.dataStore.GetENIs() < (c.maxENI - c.unmanagedENI - reserveSlotForTrunkENI) {
+		if c.hasRoomForEni() {
 			if err = c.tryAllocateENI(ctx); err == nil {
 				c.updateLastNodeIPPoolAction()
 			} else {
 				// Note that no error is returned if ENI allocation fails. This is because ENI allocation failure should not cause node to be "NotReady".
 				log.Debugf("Error trying to allocate ENI: %v", err)
 			}
 		} else {
-			log.Debugf("Skipping ENI allocation as the max ENI limit of %d is already reached (accounting for %d unmanaged ENIs and %d trunk ENIs)",
-				c.maxENI, c.unmanagedENI, reserveSlotForTrunkENI)
+			log.Debugf("Skipping ENI allocation as the max ENI limit is already reached")
```

Comment:
I considered that, and maybe it would be good to print in `hasRoomForEni`, but in general it should be easy enough to find that I figured the overhead was not worth it